File: Juego/Invasiones/fuente/GUI/CajaGUI.cs
using System;
using System.Collections.Generic;
using System.Text;
using Invasiones.Dibujo;

namespace Invasiones.GUI
{
	public abstract class CajaGUI
	{

		/// <summary>
		/// Posicion en donde se va a pintar el menu de confirmación.
		/// </summary>
		protected int m_y;

		/// <summary>
		/// Posicion en donde se va a pintar el menu de confirmación.
		/// </summary>
		protected int m_x;

		/// <summary>
		/// La fuente utilizada para escribir el nombre del boton.
		/// </summary>
		protected Fuente m_fuente;

		/// <summary>
		/// El ancho del boton. Es el ancho de la imagen, si la imagen esta seteada.
		/// </summary>
		protected int m_ancho;

		/// <summary>
		/// El alto del boton. Es el alto de la imagen, si la imagen esta seteada.
		/// </summary>
		protected int m_alto;

		/// <summary>
		/// La imagen de fondo del botón, cuando no esta el puntero del mouse sobre la imagen.
		/// </summary>
		protected Superficie m_imagen;

		/// <summary>
		/// El nombre del botón.
		/// </summary>
		protected int m_leyenda;

		/// <summary>
		/// Setea la posicion de la caja en x, y tomados desde el ancla.
		/// </summary>
		/// <param name="x">La posicion en x.</param>
		/// <param name="y">La posicion en y.</param>
		/// <param name="ancla">El ancla desde donde tomar el x y el y</param>
		public abstract void SetearPosicion(int x, int y, int ancla);

		/// <summary>
		/// Actualiza segun los eventos de entrada.
		/// </summary>
		public abstract int Actualizar();

		/// <summary>
		/// Dibuja la caja
		/// </summary>
		public abstract void Dibujar(Video g);

		/// <summary>
		/// Devuelve el alto de la caja.
		/// </summary>
        public int Alto
        {
            get
            {
                return m_alto;
            }
        }

		/// <summary>
		/// Devuelve el ancho de la caja.
		/// </summary>
        public int Ancho
        {
            get
            {
                return m_ancho;
            }
        }

	}
}

File: macOS/Invasiones/Invasiones/GUI/CajaGUI.swift
//
//  CajaGUI.swift
//  Invasiones
//
//  Created by Lucia Medina Fretes on 06.04.26.
//
//  Port of CajaGUI.cs — abstract base class for all GUI components.
//

import Foundation

class CajaGUI {

    // MARK: - Declarations
    var m_y:       Int = 0
    var m_x:       Int = 0
    var m_fuente:  Fuente?
    var m_ancho:   Int = 0
    var m_alto:    Int = 0
    var m_imagen:  Superficie?
    var m_leyenda: Int = 0

    // MARK: - Properties
    var alto:  Int { m_alto  }
    var ancho: Int { m_ancho }

    // MARK: - Initializer
    init() {}

    // MARK: - Abstract methods
    func setearPosicion(_ x: Int, _ y: Int, _ ancla: Int) {
        fatalError("\(type(of: self)).setearPosicion must be overridden")
    }

    @discardableResult
    func actualizar() -> Int {
        fatalError("\(type(of: self)).actualizar must be overridden")
    }

    func dibujar(_ g: Video) {
        fatalError("\(type(of: self)).dibujar must be overridden")
    }
}
